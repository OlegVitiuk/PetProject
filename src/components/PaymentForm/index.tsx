import { pay } from "actions/paymentActions";
import axios from "axios";
import * as React from "react";
import { connect } from "react-redux";
import StripeCheckout from "react-stripe-checkout";
import { bindActionCreators } from "redux";
import { Action } from "redux";
import { ThunkDispatch } from "redux-thunk";

interface LinkDispatchProps {
  pay: (goalId: string) => void;
}

interface PaymentFormProps {
  goalId: string;
  goalName: string;
}

type Props = LinkDispatchProps & PaymentFormProps;

interface PaymentState {}

class PaymentForm extends React.Component<Props, PaymentState> {
  private publicStrapiKey = "pk_test_hHVCjlpDhyjqd85J9wTRyslE00lIxUPRR0";

  public handleToken = (token: any): void => {
    const { pay, goalId, goalName } = this.props;

    const defaultGoalPrice = 100 * 100;
    const paymentData = {
      name: goalName,
      price: defaultGoalPrice
    };

    console.log(token, "token");
    axios
      .post("http://localhost:8080/checkout", {
        token,
        data: paymentData
      })
      .then(res => {
        const { status } = res.data;
        console.log("Response:", res.data);

        if (status === "success") {
          pay(goalId);
        } else {
          console.error("something happened");
        }
      });
  };

  public render() {
    return (
      <StripeCheckout
        stripeKey={this.publicStrapiKey}
        token={this.handleToken}
        amount={100 * 100}
        name="GOAL"
        billingAddress={true}
        shippingAddress={true}
      />
    );
  }
}

export default connect(
  null,
  (dispatch: ThunkDispatch<any, any, Action>): LinkDispatchProps => ({
    pay: bindActionCreators(pay, dispatch)
  })
)(PaymentForm);
