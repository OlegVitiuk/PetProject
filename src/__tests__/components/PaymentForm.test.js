import { shallow } from 'enzyme';
import React from 'react';
import PaymentForm from 'components/PaymentForm';
import sinon from 'sinon';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const store = configureStore([thunk])();

let [pay] = new Array(1).fill(jest.fn());

function shallowSetup() {
  // Sample props to pass to our shallow render
  const props = {
    pay,
    goalId: 'sdagdsg',
    goalName: 'asfdasf'
  };
  // wrapper instance around rendered output
  const enzymeWrapper = shallow(
    <PaymentForm {...props} store={store} />
  ).dive();

  return {
    props,
    enzymeWrapper
  };
}

describe('Payment form', () => {
  it('should render Payment form', () => {
    // Setup wrapper and assign props.
    const { enzymeWrapper, props } = shallowSetup();

    expect(enzymeWrapper.exists()).toBe(true);
    expect(enzymeWrapper).toMatchSnapshot();
    expect(enzymeWrapper.find('div').prop('data-test')).toEqual('paymentForm');
  });

  it('should call handleToken', () => {});
});
