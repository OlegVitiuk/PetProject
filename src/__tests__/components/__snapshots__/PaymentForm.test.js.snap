// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment form should render Payment form 1`] = `ShallowWrapper {}`;
